server/nuxeo-launcher/src/main/java/org/nuxeo/launcher/NuxeoLauncher.java

```diff
@@ -1695,131 +1596,235 @@ protected void setConfigProperties() throws ConfigurationException, IOException,
+            throw new NuxeoLauncherException("Could not start process: " + e.getMessage(),
+                    strict ? EXIT_CODE_NOT_CONFIGURED : EXIT_CODE_ERROR, e);
         }
     }
 
     /**
-     * @return true if Nuxeo is ready
+     * Do not directly call this method without a call to {@link #checkNoRunningServer()}
+     *
+     * @see #doStart()
+     * @throws IOException in case of issue with process.
      */
-    protected boolean waitForEffectiveStart() throws InterruptedException {
-        long startTime = new Date().getTime();
+    protected Process start(boolean logProcessOutput) throws IOException {
+        // build command to start nuxeo
+        List<String> startCommand = new ArrayList<>();
+        startCommand.add(getJavaExecutable().getPath());
+        startCommand.addAll(getJavaOptsProperty(Function.identity()));
+        startCommand.add("-cp");
+        startCommand.add(getClassPath());
+        startCommand.addAll(getNuxeoProperties());
+        startCommand.addAll(getServerProperties());
+        if (strict) {
+            startCommand.add("-Dnuxeo.start.strict=true");
+        }
+
+        startCommand.add(TomcatConfigurator.STARTUP_CLASS);
+        startCommand.add("start");
+        startCommand.addAll(Arrays.asList(params));
+
+        // build and start process
+        ProcessBuilder pb = new ProcessBuilder(getOSCommand(startCommand));
+        pb.directory(configurationGenerator.getNuxeoHome());
+        if (logProcessOutput) {
+            // don't redirect input as we want a graceful shutdown
+            pb.redirectOutput(ProcessBuilder.Redirect.INHERIT).redirectError(ProcessBuilder.Redirect.INHERIT);
+        }
+        log.debug("Server command: {}", pb::command);
+        Process nuxeoProcess = pb.start();
+        nuxeoProcess.onExit().thenAccept(p -> {
+            if (SystemUtils.IS_OS_WINDOWS && configurationGenerator.getNuxeoHome().getPath().contains(" ")) {
+                // NXP-17679
+                log.error("The server path must not contain spaces under Windows.");
+            }
+            int exitValue = p.exitValue();
+            if (exitValue != 0) {
+                log.error("Server stopped with status: {}", exitValue);
+            }
+        });
+
+        // get pid and write it to the disk for later use
+        String pid;
+        try {
+            pid = String.valueOf(nuxeoProcess.pid());
+        } catch (UnsupportedOperationException e) {
+            log.warn("Unable to get process ID from process: {}, please report it to Nuxeo", nuxeoProcess);
+            // fallback on process manager
+            pid = processManager.findPid(processRegex).orElse(null);
+        }
+        if (pid == null) {
+            log.warn("Sent server start command but could not get process ID.");
+        } else {
+            log.warn("Server started with process ID {}.", pid);
```

Comment:
If you could remove this trailing "." after the PID, it would finally be possible to copy the PID directly with the mouse, this will be a major UX improvement!